src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -763,3 +763,31 @@ function lollipop_graph(n1::T, n2::T) where {T <: Integer}
     add_edge!(g, n1, n1+1)
     return g
 end
+
+"""
+    circulant_graph(n, connection_set)
+
+Create a [circulant graph](https://en.wikipedia.org/wiki/Circulant_graph) with `n` vertices with connection set as `connection_set`.
+
+### Implementation Notes
+`n` must be at least 1 so that the graph has at least one vertex.
+
+# Examples
+```jldoctest
+julia> circulant_graph(2, [1])
+{2, 1} undirected simple Int64 graph
+
+julia> circulant_graph(Int8(3), [Int8(1)])
+{3, 3} undirected simple Int8 graph
+```
+"""
+function circulant_graph(n::T, connection_set::Vector{T}) where {T <: Integer}
+    (n < 1) && throw(DomainError("n=$n must be at least 1"))
+    g = SimpleGraph(n)
+
+    @inbounds for u = 1:n, v = 1:n
+        (mod(v - u, n) in connection_set || mod(u - v, n) in connection_set) && add_edge!(g, u, v)
+    end
```

Comment:
this is a clever use of shortcuts, but we don't allow shortcuts with mutating predicates (https://github.com/JuliaGraphs/LightGraphs.jl/blob/master/CONTRIBUTING.md#development-guidelines, bullet 10).

Reply:
Also, prefer the infix notation:
```
(v - u) % n in connection_set || (u - v) % n in connection_set
```
come to think about it, one of these is going to be <=0. Why not
```
@inbounds for u = 1:n-1, v = u:n
    if (v - u) % n in connection_set
        ...
    end
end
```
?

Reply:
Thanks. I'll incorporate your suggestions about not using shortcuts for mutating predicates and infix notation.

I don't think this will work.
```julia
@inbounds for u = 1:(n - 1), v = (u + 1):n # parens just for clarity.
    if (v - u) % n in connection_set
        ...
    end
end
```
Consider `n = 3` and `connection_set = [1]`
This should be a triangle graph with edges `[(1,2); (2,3); (1,3)]`. You can check the table [here](https://mathworld.wolfram.com/CirculantGraph.html).

For u = 1 and v = 3, (v - u)%n = 2 which isn't in the connection set. So your algorithm won't add an edge between 1 and 3.

However, (u - v)%n = (-2)%3 = (-2 + 3) % 3 = 1 which is in the connection set.

The below snippet will work
```julia
@inbounds for u = 1:(n - 1), v = (u + 1):n # parens just for clarity.
    if (v - u) % n in connection_set || (u - v + n) % n in connection_set
        ...
    end
end
```

or we could switch back to `mod(u - v, n)` if we want to avoid adding `+n`. Note that:

```Julia
julia> mod(-2, 3)
1

julia> -2 % 3
-2
```

So something like the below snippet would also work:

```Julia
@inbounds for u = 1:(n - 1), v = (u + 1):n # parens just for clarity.
    if mod(v - u, n) in connection_set || mod(u - v, n) in connection_set
        ...
    end
end
```

Reply:
Yes, I discounted the difference between `mod` and `%` for negative numbers. Use `mod` and the modified loop, since we will then cut the number of iterations by half.

Sorry for taking you down a bad path.